vector/src/main/java/im/vector/riotx/features/crypto/verification/VerificationBottomSheetViewModel.kt

```diff
@@ -334,38 +341,80 @@ class VerificationBottomSheetViewModel @AssistedInject constructor(
             }
             is VerificationAction.GotResultFromSsss            -> {
-                try {
-                    action.cypherData.fromBase64().inputStream().use { ins ->
-                        val res = session.loadSecureSecret<Map<String, String>>(ins, action.alias)
-                        val trustResult = session.cryptoService().crossSigningService().checkTrustFromPrivateKeys(
-                                res?.get(MASTER_KEY_SSSS_NAME),
-                                res?.get(USER_SIGNING_KEY_SSSS_NAME),
-                                res?.get(SELF_SIGNING_KEY_SSSS_NAME)
-                        )
-                        if (trustResult.isVerified()) {
-                            // Sign this device and upload the signature
-                            session.sessionParams.credentials.deviceId?.let { deviceId ->
-                                session.cryptoService()
-                                        .crossSigningService().trustDevice(deviceId, object : MatrixCallback<Unit> {
-                                            override fun onFailure(failure: Throwable) {
-                                                Timber.w(failure, "Failed to sign my device after recovery")
-                                            }
-                                        })
-                            }
+                handleSecretBackFromSSSS(action)
+            }
+        }.exhaustive
+    }
 
-                            setState {
-                                copy(verifiedFromPrivateKeys = true)
-                            }
-                        } else {
-                            // POP UP something
-                            _viewEvents.post(VerificationBottomSheetViewEvents.ModalError("Failed to import keys"))
-                        }
+    private fun handleSecretBackFromSSSS(action: VerificationAction.GotResultFromSsss) {
+        try {
+            action.cypherData.fromBase64().inputStream().use { ins ->
+                val res = session.loadSecureSecret<Map<String, String>>(ins, action.alias)
+                val trustResult = session.cryptoService().crossSigningService().checkTrustFromPrivateKeys(
+                        res?.get(MASTER_KEY_SSSS_NAME),
+                        res?.get(USER_SIGNING_KEY_SSSS_NAME),
+                        res?.get(SELF_SIGNING_KEY_SSSS_NAME)
+                )
+                if (trustResult.isVerified()) {
+                    // Sign this device and upload the signature
+                    session.sessionParams.credentials.deviceId?.let { deviceId ->
+                        session.cryptoService()
+                                .crossSigningService().trustDevice(deviceId, object : MatrixCallback<Unit> {
+                                    override fun onFailure(failure: Throwable) {
+                                        Timber.w(failure, "Failed to sign my device after recovery")
```

Comment:
Should we notify the user if this fails?

Reply:
It should not fail :D 
Not sure how to present that to user, for now I just log it to help debugging